components/application-deployers/org.wso2.carbon.application.deployer.synapse/src/main/java/org/wso2/carbon/application/deployer/synapse/SynapseAppDeployer.java

```diff
@@ -307,11 +304,11 @@ private void deploySynapseLibrary(List<Artifact.Dependency> artifacts,
                                         String libraryPackage = artifactName.substring(1, artifactName.lastIndexOf("}"));
                                         updateStatus(artifactName, libName, libraryPackage, ServiceBusConstants.ENABLED, axisConfig);
                                     }
+                                } else {
+                                    log.error("The qualified name of the synapse library is empty.");
```

Comment:
Any reason why don't we throws exception here ? If that's because you need to continue CApp deployment even some library deployment fails ?, then better add a comment and explain it. Else throws the exception.